pip_review/__main__.py

```diff
@@ -245,6 +248,20 @@ def main():
     elif args.raw:
         for pkg in outdated:
             logger.info('{0}=={1}'.format(pkg['name'], pkg['latest_version']))
+    elif args.tabulated:
+        selected = []
+        logger.info('{:30}{:20}{:20}'.format('name','latest version','version'))
+        logger.info('{:30}{:20}{:20}'.format('-'*29,'-'*19,'-'*19))
+        for pkg in outdated:
+            logger.info('{:30}{:20}{:20}'.format(
+                pkg['name'], pkg['latest_version'], pkg['version']
+            ))
+            if args.interactive:
+                answer = ask_to_install()
+                if answer in ['y', 'a']:
+                    selected.append(pkg)
+        if selected:
+            update_packages(selected, install_args, args.continue_on_fail, args.freeze_outdated_packages)
```

Comment:
This insertion looks like you copied the `else` clause below and then made modifications to sculpt the behavior into something more to your liking. Which is totally understandable! However, there are some problems with this:

- Copying existing code and then modifying always introduced the problem that you are repeating a pattern. If you change the pattern in one place at some later time, you have to choose between updating all other copies of the pattern as well, or accepting that your program has become less consistent. The latter option is often a mistake, if you are changing the pattern because of a bug. Keep it DRY: Don't Repeat Yourself! (Or somebody else!)
- However, in this case I think the pattern you copied does not suit your goal very well, anyway. If `args.interactive` is true, the interactive prompts will interlace with the table rows, ruining your beautiful table. If somebody passes `--table --interactive` on the command line, I think they will rather expect `pip-review` to *first* print out the entire table and then *also* ask every package one by one, as is currently happening in the `else` clause.
- In addition, the pattern you copied did not need to align subsequent lines, so it could do everything in one loop. By copying this single loop, you forced yourself to hard-code the column widths, which is not great. Most of the time you are reserving too much space, while sometimes a long package name will still derail things. It would be much better to base the columns widths one the widest values in the actual data.